### ISSUE TITLE
addframework   failed in xcode7 

### ISSUE THREAD

**Issue by GangWang** [appsquickly/XcodeEditor#56]
I updated xcode to xcode7,   happened this

static const NSString* SDK_PATH = @"/Applications/Xcode.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/SDKs/iPhoneOS.sdk";

+ (NSString*)libzDylibPath
{
//    return @"/usr/lib/libz.dylib";
    return [SDK_PATH stringByAppendingPathComponent:@"/usr/lib/libz.tbd"];
}

XCFrameworkDefinition* frameworkDefinition = [[XCFrameworkDefinition alloc] initWithFilePath:[XCFrameworkPath libzDylibPath] copyToDestination:NO];
    
[frameworksGroup addFramework:frameworkDefinition toTargets:[project targets]];

![b33cc970-f502-4c8b-b1bb-b0c65d3d465a](https://cloud.githubusercontent.com/assets/1000877/10064758/a6a404fe-62ae-11e5-8e5e-583db718026b.png)

here is ok~   but when i build the project,  failed for this

![fffc1340-e6e8-4e00-89a1-fdc3995c95b2](https://cloud.githubusercontent.com/assets/1000877/10064775/ce32785c-62ae-11e5-8aad-68fe698bae6a.png)

**Comment by cezheng** [comment]
linking against *.tbd libraries seems to be an unfinished feature in xcode 7, try adding -lz in the OTHER_LD_FLAGS which will do what you want

**Comment by GangWang** [comment]
i wish the author will finish link *.tbd libraries feature  in  xcode 7    soon， ;)

**Comment by cezheng** [comment]
I guess the tbd issue is not XcodeEditor's problem but a problem of Xcode itself, so I guess this issue should be closed

**Comment by GangWang** [comment]
the reason is the pbxproj file format is changed, may be need to support it in  XcodeEditor

**Comment by cezheng** [comment]
I don't think so. Even if you manually add the libz.tbd to your project in Xcode, it fails

**Comment by GangWang** [comment]
i manually add the libz.tbd to my project in Xcode, it is ok  
![195d1538-56f5-4ff5-8148-de7813b7cbf9](https://cloud.githubusercontent.com/assets/1000877/10067477/8f5a7b9a-62cf-11e5-94e6-7fe00e326be4.png)

**Comment by cezheng** [comment]
Oh, maybe *.tbd cannot be used in a static library project. I have a static library project and if I add a tbd file the build just fails

**Comment by jasperblues** [comment]
@cezheng @GangWang Agree we can close this?

**Comment by akame** [comment]
I think this problem because XcodeEditor doesn't support .tbd file,
I add Dylib = 24 in XcodeSourceFileType.h 
and  @"sourcecode.text-based-dylib-definition" : @(Dylib)  in   XcodeSourceFileType.m(NSDictionaryWithXCFileReferenceTypes). 

At last, I copy a new function from function :- (void)addFramework:(XCFrameworkDefinition *)frameworkDefinition; in XCGourp.m .
I change:
fileReference = [self makeFileReferenceWithPath:path name:name type:Framework] 
to 
fileReference = [self makeFileReferenceWithPath:path name:name type:Dylib]

I'm a new programmer ,and if I was wrong, forgive me .

**Comment by GangWang** [comment]
@akame   hahaha~~~~~       thank you!    you are right, i follow u step, it works!      @jasperblues

**Comment by forwardto9** [comment]
@akame nice! but you also add buidPharse item
